presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergLogicalPlanner.java

```diff
@@ -77,6 +102,137 @@ protected QueryRunner createQueryRunner()
+                output(values("linenumber")));
+
+        assertPlan(pushdownFilterEnabled, "SELECT linenumber FROM lineitem WHERE orderkey > 10 AND cardinality(NULL) > 0",
+                output(values("linenumber")));
+
+        // Remaining predicate is always FALSE
+        assertPlan(pushdownFilterEnabled, "SELECT linenumber FROM lineitem WHERE cardinality(ARRAY[1]) > 1",
+                output(values("linenumber")));
+
+        assertPlan(pushdownFilterEnabled, "SELECT linenumber FROM lineitem WHERE orderkey > 10 AND cardinality(ARRAY[1]) > 1",
+                output(values("linenumber")));
+
+        // TupleDomain predicate is always FALSE
+        assertPlan(pushdownFilterEnabled, "SELECT linenumber FROM lineitem WHERE orderkey = 1 AND orderkey = 2",
+                output(values("linenumber")));
+
+        assertPlan(pushdownFilterEnabled, "SELECT linenumber FROM lineitem WHERE orderkey = 1 AND orderkey = 2 AND linenumber % 2 = 1",
+                output(values("linenumber")));
+
+        FunctionAndTypeManager functionAndTypeManager = getQueryRunner().getMetadata().getFunctionAndTypeManager();
+        FunctionResolution functionResolution = new FunctionResolution(functionAndTypeManager.getFunctionAndTypeResolver());
+        RowExpression remainingPredicate = new CallExpression(EQUAL.name(),
+                functionResolution.comparisonFunction(EQUAL, BIGINT, BIGINT),
+                BOOLEAN,
+                ImmutableList.of(
+                        new CallExpression("mod",
+                                functionAndTypeManager.lookupFunction("mod", fromTypes(BIGINT, BIGINT)),
+                                BIGINT,
+                                ImmutableList.of(
+                                        new VariableReferenceExpression(Optional.empty(), "orderkey", BIGINT),
+                                        constant(2))),
+                        constant(1)));
+
+        assertPlan(pushdownFilterEnabled, "SELECT linenumber FROM lineitem WHERE mod(orderkey, 2) = 1",
+                output(exchange(
+                        strictTableScan("lineitem", identityMap("linenumber")))),
+                plan -> assertTableLayout(
+                        plan,
+                        "lineitem",
+                        TupleDomain.all(),
+                        remainingPredicate,
+                        ImmutableSet.of("orderkey")));
+
+        assertPlan(pushdownFilterEnabled, "SELECT orderkey, linenumber FROM lineitem WHERE mod(orderkey, 2) = 1",
+                output(exchange(
+                        strictTableScan("lineitem", identityMap("orderkey", "linenumber")))),
+                plan -> assertTableLayout(
+                        plan,
+                        "lineitem",
+                        TupleDomain.all(),
+                        remainingPredicate,
+                        ImmutableSet.of("orderkey")));
+
+        // A mix of domain and remaining predicates
```

Comment:
Can you please add
1) filter with both OR and AND. e.g. 
(partkey < 2 and suppkey < 2) or (partkey > 9 and suppkey > 9)
2) With dynamic filters